### PR TITLE
Algorand PostgreSQL

### PR DESCRIPTION
**Pull request by haammar-ledger**
Support of PostgreSQL for Algorand tests

### CODE REVIEW

File: core/test/integration/WalletFixture.hpp [hunk 1]
```diff
@@ -41,27 +41,32 @@
...
 
     void SetUp() override {
         BaseFixture::SetUp();
 
-        pool = newDefaultPool(DEFAULT_TENANT, DEFAULT_PASSWORD);
+#ifdef PG_SUPPORT
+         const bool usePostgreSQL = true;
+         auto poolConfig = DynamicObject::newInstance();
+         poolConfig->putString(api::PoolConfiguration::DATABASE_NAME, "postgres://localhost:5432/test_db");
+         pool = newDefaultPool("postgres", "", poolConfig, usePostgreSQL);
```
Comment (gagbo) [inline core/test/integration/WalletFixture.hpp:52]: can you just change the pool name ("postgres") to be more unique/random ? That will help parallelize `ctest` later

Comment (haammar-ledger) [reply core/test/integration/WalletFixture.hpp]: Done, added a 10 characters random string.

**Comment by llehmann-ledger** [comment]
Looks like it solve the issue with Cosmos but now tests in `AlgorandAccountTest` don't pass ? (at least on my machine)

**Comment by haammar-ledger** [comment]
@llehmann-ledger What's the issue? On my side tests also fail with PG so far, I have this error for all of them:
```
C++ exception with description "Cannot establish connection to the database.
fe_sendauth: no password supplied
" thrown in SetUp().
```
I didn't figure out why yet.

**Comment by llehmann-ledger** [comment]
> @llehmann-ledger What's the issue? On my side tests also fail with PG so far, I have this error for all of them:
> 
> ```
> C++ exception with description "Cannot establish connection to the database.
> fe_sendauth: no password supplied
> " thrown in SetUp().
> ```
> 
> I didn't figure out why yet.

After the last commit randomizing the pool name I got : 
```
[ RUN      ] AlgorandAccountTest.algosBalanceHistory
NOTICE:  drop cascades to constraint ethereum_operations_transaction_uid_fkey on table ethereum_operations
zsh: abort      ./core/test/algorand/build/ledger-core-algorand-tests
```

And if I remove the last commit and just add the `<api/PoolConfiguration.hpp>` include, I got : 
```
unknown file: Failure
C++ exception with description "Wallet 'algorand' for currency 'algorand' already exists" thrown in SetUp().
```

Don't know why yet, I ran the cosmos tests before I'll try to see if it does the same when running algorand tests before

**Pull request merged**